### PR TITLE
Handle namespaces and declare() statements when combining files

### PR DESCRIPTION
**Pull request by leofeyer**
This is a follow-up PR on #730, which uses the tokenizer to handle everything (PHP tags, declare statements and namespaces).

### CODE REVIEW

File: tests/Config/Loader/PhpFileLoaderTest.php [hunk 1]
```diff
@@ -93,14 +98,83 @@ public function testLoad()
...
+EOF;
+
+        $this->assertEquals(
+            $expects,
+            $this->loader->load($this->getRootDir().'/vendor/contao/test-bundle/Resources/contao/dca/tl_declare.php')
+        );
+
+        $expects = <<<'EOF'
+
+//declare(strict_types  =   1,ticks = 1);
```
Comment (Toflar) [inline tests/Config/Loader/PhpFileLoaderTest.php]: This is not the correct behaviour. It needs to be `declare(ticks=1);`. Only strip `strict_types=1`.

Comment (leofeyer) [inline tests/Config/Loader/PhpFileLoaderTest.php]: You are right.

**Comment by leofeyer** [comment]
> all my code is gone and all my commits/work disappeared (which somehow does bother me, yes)

You said "feel free to adjust them if you want to migrate over". It then got pretty complex and confusing with the PHP tokenizer, so I decided to use the `php-token-stream` library.

Now how do we solve this situation?

**Comment by Toflar** [comment]
> Now how do we solve this situation?

Well you could've PR'ed to my PR. Even if you change the whole implementation this would've automatically alerted you that the tests fail because I had test cases for all the `declare()` situations ;)

**Comment by leofeyer** [comment]
Done: https://github.com/Toflar/core-bundle/pull/6

File: tests/Config/Loader/PhpFileLoaderTest.php (hunk 1 above)
Comment (leofeyer) [inline tests/Config/Loader/PhpFileLoaderTest.php]: I have fixed this in https://github.com/Toflar/core-bundle/pull/6, which uses your tests for the `declare()` stuff.

**Pull request closed without merging**